samcli/commands/pipeline/init/cli.py

```diff
@@ -0,0 +1,43 @@
+"""
+CLI command for "pipeline init" command
+"""
+from typing import Any, Optional
+
+import click
+
+from samcli.cli.cli_config_file import configuration_option, TomlProvider
+from samcli.cli.main import pass_context, common_options as cli_framework_options
+from samcli.commands.pipeline.init.interactive_init_flow import do_interactive
+from samcli.lib.telemetry.metric import track_command
+
+SHORT_HELP = "Generates a CI/CD pipeline configuration file."
+HELP_TEXT = """
+Generates a CI/CD pipeline configuration file for a chosen CI/CD provider like Jenkins, Gitlab and Github Actions
+"""
+
+
+@click.command("init", help=HELP_TEXT, short_help=SHORT_HELP)
+@configuration_option(provider=TomlProvider(section="parameters"))
+@cli_framework_options
+@pass_context
+@track_command  # pylint: disable=R0914
+def cli(
+    ctx: Any,
+    config_env: Optional[str],
+    config_file: Optional[str],
+) -> None:
+    """
+    `sam pipeline init` command entry point
+    """
+
+    # Currently we support interactive mode only, i.e. the user doesn't provide the required arguments during the call
```

Comment:
Are there any challenges to supporting non-interactive mode?

Reply:
We discussed before and agreed that it won't be included in the first run. Actually, we are quite unsure about the use-case. We believe this command will mainly be used in an interactive way. One use-case for the non-interactive mode is the IDEs if we are integrating with it, but this integration is not discussed.